modules/pokemon.py

```diff
@@ -1504,6 +1504,8 @@ def get_opponent() -> Pokemon:
 
 last_opid = pack_uint32(0)  # ReadSymbol('gEnemyParty', size=4)
 
+class BattleTypeFlag(Flag, boundary=KEEP):
```

Comment:
Genuine question, what does `boundary=KEEP` do?

Reply:
that lets me only add the Flags we are interested to the Enum.
without the KEEP it would error, because there are other flags set.

now reading it again.. we could probably just take `IntFlag` instead of `Flag`. ther KEEP is the default behaviour.

> FlagBoundary controls how out-of-range values are handled in Flag and its subclasses.
> KEEP: Out-of-range values are kept, and the Flag membership is kept. This is the default for [IntFlag](https://docs.python.org/3/library/enum.html#enum.IntFlag):
[source](https://docs.python.org/3/library/enum.html#enum.FlagBoundary.KEEP)

i dont know if there are any performance differences between them...